### PR TITLE
[Content] MemoryRouter reference page

### LINKED ISSUE #472: [content] Add HashRouter and MemoryRouter to Solid-Router reference section
Under the heading `Alternative Routers` we nee to add both `MemoryRouter` and `HashRouter` to the references section for Solid-Router

### PR DESCRIPTION
**Pull request by birkskyum**
Closes #472
- #472

While creating this, I believe I might have uncovered a bug that makes the MemoryHistory of little use.
- https://github.com/solidjs/solid-router/issues/393

I tested it quite extensively to understand it in depth, and in short what I found is that the `set` function used to navigate to a new route doesn't work well.

I've not had the bug confirmed yet, but it's likely users of MemoryRouter will find themselves debugging almost immediately.

I wrote the Reference as if the functionality is working as intended, and I haven't added a caution box.

Request for review: @LadyBluenotes

### CODE REVIEW

**Comment by birkskyum** [comment]
I don't know why these workflow fails, since I can read the logs - so if someone could paste the logs here, or tell me what to do to make them pass (or handle it for me if easier), I'd appreciate it.

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (LadyBluenotes) [inline src/routes/solid-router/reference/components/memory-router.mdx:6]:
```suggestion
The `MemoryRouter` can be used to route while keeping the entire routing history within internal memory.
Other routers keep history externally such as in the url like the [`<HashRouter>`](/solid-router/reference/components/hash-router) or in the browser history like [`<Router>`](/solid-router/reference/components/memory-router).
```

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (LadyBluenotes) [inline src/routes/solid-router/reference/components/memory-router.mdx:9]:
```suggestion
Keeping the history in memory is useful when you want full control over the routing history.
Since `MemoryRouter` can manipulate `MemoryHistory`, it is often used for testing purposes.
```

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (LadyBluenotes) [inline src/routes/solid-router/reference/components/memory-router.mdx:14]:
```suggestion
```

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (LadyBluenotes) [inline src/routes/solid-router/reference/components/memory-router.mdx:45]:
```suggestion
In this example, a history object is pre-filled to navigate to the `/about` route, which is then passed to the `MemoryRouter`.

## Manipulating the history
```

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (LadyBluenotes) [inline src/routes/solid-router/reference/components/memory-router.mdx:47]: Can you give a quick blurb about what the functions intend to do? Ie. "To give you control over how you navigate your app" or whatever makes the most sense?

```suggestion
The `MemoryHistory` object contains functions that are used to control navigation. 
```

Comment (LadyBluenotes) [reply src/routes/solid-router/reference/components/memory-router.mdx:47]: I'm also thinking these are better presented in the form of a table vs a section for each - do you mind putting them into one?

**Review by LadyBluenotes — CHANGES_REQUESTED** [review]

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (birkskyum) [reply src/routes/solid-router/reference/components/memory-router.mdx:6]: I can add this link, just note it'll break CI until the HashRouter page also is merged.

Comment (LadyBluenotes) [reply src/routes/solid-router/reference/components/memory-router.mdx]: I'll merge them both when they're good to go :)

File: src/routes/solid-router/reference/components/memory-router.mdx
Comment (birkskyum) [reply src/routes/solid-router/reference/components/memory-router.mdx]: This was a great idea.

I added a table, and some text about each of the three types of functions in the MemoryHistory.

**Review by LadyBluenotes — APPROVED** [review]

**Pull request merged**
